package/kernel/qca-ssdk/patches/100-malibu-phy-add-support-for-manual-define-of-first-ph.patch

```diff
@@ -0,0 +1,131 @@
+set the correct bitmap...
+
+Again world is a bad place and reality is that this cause wonderful
+regression in some case of by extreme luck the first ever connected
+port working and the rest of the switch dead.
+
+The problem has been bisected to all the device that doesn't have the
+PORT1 declared in any of the bitmap.
+
+With this perfection in mind, on to the REAL problem.
+
+malibu_phy_hw_init FOR SOME REASON, set a global variable first_phy_addr
+to the first detected PHY addr that coincidentally is always PORT1.
+PORT1 addr is 0x0. The entire code in malibu_phy use this variable to
+derive the phy addrs in some function.
+
+Declaring a bitmap where the PORT1 is missing (or worse PORT4 the only
+one connected) result in first_phy_addr set to 1 or whatever phy addr is
+detected first setting wrong value all over the init stage.
+
+To fix this, introduce a new binding malibu_first_phy_addr to menually
+declare the first phy that the malibu PHY driver should use and permit
+to detach it from port bmp detection. The legacy detection is kept for
+compatibility reason.
+
+Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
+---
+ include/init/ssdk_dts.h  |  1 +
+ include/init/ssdk_init.h |  2 ++
+ src/hsl/phy/malibu_phy.c |  5 +++++
+ src/init/ssdk_dts.c      | 15 +++++++++++++++
+ 4 files changed, 23 insertions(+)
+
+--- a/include/init/ssdk_dts.h
++++ b/include/init/ssdk_dts.h
+@@ -146,6 +146,7 @@ a_uint32_t ssdk_wan_bmp_get(a_uint32_t d
+ sw_error_t ssdk_lan_bmp_set(a_uint32_t dev_id, a_uint32_t lan_bmp);
+ sw_error_t ssdk_wan_bmp_set(a_uint32_t dev_id, a_uint32_t wan_bmp);
+ a_uint32_t ssdk_inner_bmp_get(a_uint32_t dev_id);
++a_uint32_t ssdk_malibu_first_phy_addr_get(a_uint32_t dev_id);
+ hsl_reg_mode ssdk_switch_reg_access_mode_get(a_uint32_t dev_id);
+ void ssdk_switch_reg_map_info_get(a_uint32_t dev_id, ssdk_reg_map_info *info);
+ a_uint32_t ssdk_switch_pcie_base_get(a_uint32_t dev_id);
+--- a/include/init/ssdk_init.h
++++ b/include/init/ssdk_init.h
+@@ -194,6 +194,7 @@ enum ssdk_port_wrapper_cfg {
+ 		a_uint32_t lan_bmp;
+ 		a_uint32_t wan_bmp;
+ 		a_uint32_t inner_bmp;
++		a_uint32_t malibu_first_phy_addr;
+ 	} ssdk_port_cfg;
+ 
+ 	typedef struct
+@@ -384,6 +385,7 @@ ssdk_hsl_access_mode_set(a_uint32_t dev_
+ 
+ a_uint32_t ssdk_dt_global_get_mac_mode(a_uint32_t dev_id, a_uint32_t index);
+ a_uint32_t ssdk_dt_global_set_mac_mode(a_uint32_t dev_id, a_uint32_t index, a_uint32_t mode);
++a_uint32_t ssdk_malibu_first_phy_addr_get(a_uint32_t dev_id);
+ 
+ a_uint32_t
+ qca_hppe_port_mac_type_get(a_uint32_t dev_id, a_uint32_t port_id);
+--- a/src/hsl/phy/malibu_phy.c
++++ b/src/hsl/phy/malibu_phy.c
+@@ -1945,6 +1945,11 @@ static int malibu_phy_api_ops_init(void)
+ int malibu_phy_init(a_uint32_t dev_id, a_uint32_t port_bmp)
+ {
+ 	static a_uint32_t phy_ops_flag = 0;
++	a_uint32_t malibu_first_phy_addr;
++
++	malibu_first_phy_addr = ssdk_malibu_first_phy_addr_get(dev_id);
++	if (malibu_phy_addr != MAX_PHY_ADDR)
```

Comment:
You probably meant `if (malibu_first_phy_addr != MAX_PHY_ADDR)` here, as otherwise it will fail compiling